tianshou/data/batch.py

```diff
@@ -116,15 +115,17 @@ def _parse_value(v: Any):
     if isinstance(v, (list, tuple, np.ndarray)):
         if not isinstance(v, np.ndarray) and \
                 all(isinstance(e, torch.Tensor) for e in v):
-            v = torch.stack(v)
-            return v
-        v_ = _to_array_with_correct_type(v)
-        if v_.dtype == np.object and _is_batch_set(v):
-            v = Batch(v)  # list of dict / Batch
-        else:
-            # normal data list (main case)
-            # or actually a data list with objects
-            v = v_
+            return torch.stack(v)
+        try:
+            v_ = _to_array_with_correct_type(v)
```

Comment:
It fixes a bug of `Batch`:

```
>>> Batch(a=[np.zeros((3, 2)), np.zeros((3, 3))])
Batch(
    a: [array([[0., 0.],
              [0., 0.],
              [0., 0.]]),
        array([[0., 0., 0.],
              [0., 0., 0.],
              [0., 0., 0.]])],
)
```

is now working properly.

Reply:
But how about `Batch(a=[torch.zeros((3, 2)), torch.zeros((3, 3))])`?

Reply:
Now:
```
>>> Batch(a=[torch.zeros((3, 2)), torch.zeros((3, 3))])                                                                                                                                                 
Batch(
    a: array([tensor([[0., 0.],
               [0., 0.],
               [0., 0.]]),
              tensor([[0., 0., 0.],
               [0., 0., 0.],
               [0., 0., 0.]])], dtype=object),
)
```